warehouse/cli/tuf.py

```diff
@@ -0,0 +1,139 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import click
+
+from securesystemslib.interface import generate_and_write_ed25519_keypair
+
+from warehouse.cli import warehouse
+from warehouse.packaging.utils import render_simple_detail
+from warehouse.tuf.tasks import (
+    add_hashed_targets as _add_hashed_targets,
+    bump_bin_ns as _bump_bin_ns,
+    bump_snapshot as _bump_snapshot,
+    delegate_targets_bin_bins as _delegate_targets_bin_bins,
+    init_repository as _init_repository,
+)
+
+
+@warehouse.group()  # pragma: no-branch
+def tuf():
+    """
+    Manage Warehouse's TUF state.
+    """
+
+
+@tuf.group()
+def dev():
+    """
+    TUF Development purposes commands
+    """
+
+
+@dev.command()
+@click.pass_obj
+@click.option("--name", "name_", help="The name of the TUF role for this keypair")
+@click.option("--path", "path_", help="The basename of the Ed25519 keypair to generate")
+def keypair(config, name_, path_):
+    """
+    Generate a new TUF keypair, for development purposes.
+    """
+    password = config.registry.settings[f"tuf.{name_}.secret"]
+    generate_and_write_ed25519_keypair(password, filepath=path_)
+
+
+@dev.command()
+@click.pass_obj
+def new_repo(config):
+    """
+    Initialize new TUF repository from scratch, for development purposes.
+    """
+    try:
+        request = config.task(_init_repository).get_request()
+        config.task(_init_repository).run(request)
+    except FileExistsError as err:
+        raise click.ClickException(str(err))
+
+    click.echo("Repository Initialization finished.")
+
+
+@tuf.group()
+def admin():
+    """
+    TUF Administrative commands
+    """
+
+
+@admin.command()
+@click.pass_obj
+def bump_snapshot(config):
+    """
+    Bump Snapshot metadata
+    """
+    request = config.task(_bump_snapshot).get_request()
+    config.task(_bump_snapshot).run(request)
+    click.echo("Snapshot bump finished.")
+
+
+@admin.command()
+@click.pass_obj
+def bump_bin_ns(config):
+    """
+    Bump BIN-S roles
+    """
+    request = config.task(_bump_bin_ns).get_request()
+    config.task(_bump_bin_ns).run(request)
+    click.echo("Hash-bins BIN-S Roles bump finished.")
```

Comment:
Looks like you yourself got confused here. :P Did you mean...?

```suggestion
    Bump BIN-N roles
    """
    request = config.task(_bump_bin_ns).get_request()
    config.task(_bump_bin_ns).run(request)
    click.echo("Hash-bins BIN-N Roles bump finished.")
```